src/application/proxy.ts

```diff
@@ -101,3 +106,14 @@ export function makeid(length: number): string {
   }
   return result;
 }
+
+function isMarinaEventType(str: string): str is MarinaEventType {
+  return (
```

Comment:
why not `Object.keys(initialEventListeners)` ? 

it would be hard to maintain all these strings if we add, remove events